src/components/InstallButton.js

```diff
@@ -10,21 +10,61 @@ import {
   NR1_LOGIN_URL,
   NR1_GUIDED_INSTALL_NERDLET,
   NR1_PACK_DETAILS_NERDLET,
+  UTM_PARAMETERS,
+  NR1_SIGNUP_URL,
 } from '../data/constants';
 import { quickstart } from '../types';
 
+/**
+ * @param {String} parameters
+ * @returns {Array}
+ */
+const strippedParameters = (parameters) => {
+  return parameters[0] === '?'
+    ? parameters.slice(1).split('&')
+    : parameters.split('&');
+};
+
+/**
+ * @param {Object} location
+ * @returns {String}
+ */
+const checkUtmParameters = (location) => {
+  const { state } = location;
+  if (!state) {
+    return false;
+  }
+  const { search: parameters } = state.prevPath;
+  const parametersArray = strippedParameters(parameters);
```

Comment:
[URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) will make the parsing and moving around search/query parameters much easier than dealing with the raw string.